### PR TITLE
Duration fix

### PR DESCRIPTION
**Pull request by RikkiGibson**
This allows us to tolerate usage of multiple moment versions simultaneously. Dupes work already done in Azure/azure-sdk-for-node#2453. Suggest particularly looking at the first two commits--the third is more "noise" (sourcemaps, js files)

### CODE REVIEW

File: lib/serializer.ts [hunk 1]
```diff
@@ -232,7 +232,7 @@ export class Serializer {
         }
         value = this.dateToUnixTime(value);
       } else if (typeName.match(/^TimeSpan$/ig) !== null) {
-        if (!isDuration(value)) {
+        if (!(isDuration(value) || (value.constructor && value.constructor.name === "Duration" && (value as any).isValid && (value as any).isValid()))) {
```
Comment (daschult) [inline lib/serializer.ts]: Do you need to check if isValid is a function? I'm sure you're fine not checking, but it isn't a function, will this throw an exception when you try to invoke it?

Comment (RikkiGibson) [reply lib/serializer.ts]: Can go with `typeof (value as any).isValid === 'function'`

Comment (RikkiGibson) [reply lib/serializer.ts]: I wish the moment type definitions simply included this member on Durations, maybe will just augment the type definition here locally

Comment (RikkiGibson) [reply lib/serializer.ts]: Done @daschult

Comment (daschult) [reply lib/serializer.ts]: Cool. I'm glad that worked out well.

**Pull request merged**